plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -63,16 +73,22 @@ public CachingDirectoryLister(HiveConfig hiveClientConfig)
 
     public CachingDirectoryLister(Duration expireAfterWrite, long maxSize, List<String> tables)
     {
-        this.cache = EvictableCacheBuilder.newBuilder()
+        this(Optional.of(expireAfterWrite), maxSize, tables, new FileSystemDirectoryLister());
+    }
+
+    public CachingDirectoryLister(Optional<Duration> expireAfterWrite, long maxSize, List<String> tables, DirectoryLister delegate)
+    {
+        EvictableCacheBuilder<Path, ValueHolder> cacheBuilder = EvictableCacheBuilder.newBuilder()
                 .maximumWeight(maxSize)
```

Comment:
This is confusing, we send `maxSize` parameter to `maximumWeight` but there is also a `maximumSize` in EvictableCacheBuilder.
It seems to be about the max number of file listings that will be cached, so let's rename `maxSize` to something which clarifies that.